src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/HotReload/ProjectHotReloadSessionManager.cs

```diff
@@ -374,6 +383,14 @@ private async Task<bool> StopProjectAsync(HotReloadState hotReloadState, Cancell
                     ),
                     cancellationToken);
             }
+            finally
+            {
+                // No more sessions removes the project from hot reload mode
+                if (_activeSessions.Count == 0)
+                {
+                    await _projectHotReloadNotificationService.Value.SetHotReloadStateAsync(isInHotReload: false);
+                }
```

Comment:
I think this could fire incorrectly in the case that `_activeSessions.Remove(hotReloadState.Process.Id)` returns `false`. It should only fire on the "edge" from 1 to 0, not just if it's 0 here.

Reply:
Good catch.